Patch-Source: https://github.com/Governikus/AusweisApp2/pull/48
From 67f15b602a5642bfe36b08b730a7d6624ea106d1 Mon Sep 17 00:00:00 2001
From: Tilman Vogel <tilman.vogel@web.de>
Date: Mon, 16 Oct 2023 14:03:38 +0200
Subject: [PATCH] Fix compilation failure on openSUSE Tumbleweed (10/2023)

PcscUtils: Using LONG (long) as the underlying type for enum PcscReturnCode
caused an ambiguity when resolving QDataStream operators. Therefore, now
explicitly use qint64. Before, on 64-bit platforms, a 64-bit signed int was
used and on 32-bit platforms, a 32-bit signed int was used.

TintableIcon.qml: transformOrigin is a FINAL property since Qt 6.6 such
that the hiding alias does not work anymore. Instead, use the standard
property and tie image.transformOrigin to root.transformOrigin.
---
 resources/qml/Governikus/Global/TintableIcon.qml | 2 +-
 src/card/pcsc/PcscUtils.h                        | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/resources/qml/Governikus/Global/TintableIcon.qml b/resources/qml/Governikus/Global/TintableIcon.qml
index b7e9bf0e..ec7c83d6 100644
--- a/resources/qml/Governikus/Global/TintableIcon.qml
+++ b/resources/qml/Governikus/Global/TintableIcon.qml
@@ -16,7 +16,6 @@ Item {
 	property alias sourceSize: image.sourceSize
 	property color tintColor: Style.color.primary_text
 	property bool tintEnabled: true
-	property alias transformOrigin: image.transformOrigin
 
 	implicitHeight: image.implicitHeight
 	implicitWidth: image.implicitWidth
@@ -26,6 +25,7 @@ Item {
 		anchors.fill: parent
 		asynchronous: true
 		fillMode: Image.PreserveAspectFit
+		transformOrigin: root.transformOrigin
 		layer.enabled: root.tintEnabled && GraphicsInfo.api !== GraphicsInfo.Software
 
 		layer.effect: ShaderEffect {
diff --git a/src/card/pcsc/PcscUtils.h b/src/card/pcsc/PcscUtils.h
index 992156ac..980f6a94 100644
--- a/src/card/pcsc/PcscUtils.h
+++ b/src/card/pcsc/PcscUtils.h
@@ -43,7 +43,7 @@ using PCSC_CHAR = char;
 using PCSC_CHAR_PTR = char*;
 using PCSC_CUCHAR_PTR = const uchar*;
 using PCSC_INT = DWORD;
-using PCSC_RETURNCODE = LONG;
+using PCSC_RETURNCODE = qint64;
 using PCSC_UCHAR_PTR = uchar*;
 #endif
 
